src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE_STORAGE = 10;
+    private final K[] key;
+    private final V[] value;
```

Comment:
better to give different names to the arrays than the names of the input data then you will be able to remove unnecessary "this." from the code.
```suggestion
    private final V[] differentValue;
```